### PR TITLE
Add gatekeeper flag for disabling DB queries on studio homepage

### PR DESCRIPTION
**Pull request by joshlory**
This change adds two gatekeeper flags for reducing queries on the studio homepage:
- `logged_in_studio_homepage: false` removes all queries except the user hero.
- `logged_in_studio_hero: false` also remove the user hero queries.

With DB queries disabled:

screen shot 2015-12-05 at 5 24 09 pm
## Queries

Gatekeeper flag 'logged_in_studio_homepage' set to `true`: **(un-cached queries highlighted with ***)**

``` bash
Started GET "/" for 127.0.0.1 at 2015-12-04 20:29:46 -0800
Processing by HomeController#index as HTML
  *** User Load (0.5ms)  SELECT  `users`.* FROM `users` WHERE `users`.`deleted_at` IS NULL AND `users`.`id` = 2  ORDER BY `users`.`id` ASC LIMIT 1
  Rendered layouts/_survey.html.haml (0.2ms)
  Rendered devise/confirmations/_alert.html.haml (4.7ms)
  *** UserScript Exists (0.4ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1
  *** Script Load (0.6ms)  SELECT `scripts`.* FROM `scripts` INNER JOIN `user_scripts` ON `scripts`.`id` = `user_scripts`.`script_id` WHERE `user_scripts`.`user_id` = 2 AND `scripts`.`hidden` = 0 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc
  *** UserLevel Load (0.5ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 89  ORDER BY id desc
  CACHE (0.1ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  CACHE (0.1ms)  SELECT `scripts`.* FROM `scripts` INNER JOIN `user_scripts` ON `scripts`.`id` = `user_scripts`.`script_id` WHERE `user_scripts`.`user_id` = 2 AND `scripts`.`hidden` = 0 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc  [["user_id", 2], ["hidden", false]]
  CACHE (0.1ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  CACHE (0.2ms)  SELECT `scripts`.* FROM `scripts` INNER JOIN `user_scripts` ON `scripts`.`id` = `user_scripts`.`script_id` WHERE `user_scripts`.`user_id` = 2 AND `scripts`.`hidden` = 0 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc  [["user_id", 2], ["hidden", false]]
  CACHE (0.3ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  CACHE (0.2ms)  SELECT `scripts`.* FROM `scripts` INNER JOIN `user_scripts` ON `scripts`.`id` = `user_scripts`.`script_id` WHERE `user_scripts`.`user_id` = 2 AND `scripts`.`hidden` = 0 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc  [["user_id", 2], ["hidden", false]]
  Rendered api/user_hero.html.haml (180.5ms)
  CACHE (0.1ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  CACHE (0.1ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  *** UserScript Load (0.5ms)  SELECT `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc
  *** UserScript Load (0.6ms)  SELECT `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND (user_scripts.completed_at is not null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc
  *** UserScript Load (0.4ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 45  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
  *** UserLevel Load (0.3ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 45  ORDER BY id desc
  CACHE (0.1ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 45  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1  [["user_id", 2], ["script_id", 45]]
  CACHE (0.1ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 45  ORDER BY id desc  [["user_id", 2], ["script_id", 45]]
  CACHE (0.1ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 45  ORDER BY id desc  [["user_id", 2], ["script_id", 45]]
  CACHE (0.3ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 45  ORDER BY id desc  [["user_id", 2], ["script_id", 45]]
  Rendered shared/_course_progress_block.html.haml (117.7ms)
  *** UserScript Load (0.7ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 50  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
  *** UserLevel Load (0.5ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 50  ORDER BY id desc
  CACHE (0.1ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 50  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1  [["user_id", 2], ["script_id", 50]]
  CACHE (0.1ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 50  ORDER BY id desc  [["user_id", 2], ["script_id", 50]]
  CACHE (0.1ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 50  ORDER BY id desc  [["user_id", 2], ["script_id", 50]]
  CACHE (0.1ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 50  ORDER BY id desc  [["user_id", 2], ["script_id", 50]]
  Rendered shared/_course_progress_block.html.haml (37.9ms)
  *** UserScript Load (0.4ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 46  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
  *** UserLevel Load (0.4ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 46  ORDER BY id desc
  CACHE (0.1ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 46  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1  [["user_id", 2], ["script_id", 46]]
  CACHE (0.1ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 46  ORDER BY id desc  [["user_id", 2], ["script_id", 46]]
  CACHE (0.1ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 46  ORDER BY id desc  [["user_id", 2], ["script_id", 46]]
  CACHE (0.1ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 46  ORDER BY id desc  [["user_id", 2], ["script_id", 46]]
  Rendered shared/_course_progress_block.html.haml (302.3ms)
  *** UserScript Load (0.4ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 39  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
  CACHE (0.1ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  CACHE (0.1ms)  SELECT `scripts`.* FROM `scripts` INNER JOIN `user_scripts` ON `scripts`.`id` = `user_scripts`.`script_id` WHERE `user_scripts`.`user_id` = 2 AND `scripts`.`hidden` = 0 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc  [["user_id", 2], ["hidden", false]]
  Rendered shared/_courseblock_action.haml (13.2ms)
  Rendered shared/_course_tall_block.haml (18.0ms)
  *** UserScript Load (0.4ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 40  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
  CACHE (0.1ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  CACHE (0.1ms)  SELECT `scripts`.* FROM `scripts` INNER JOIN `user_scripts` ON `scripts`.`id` = `user_scripts`.`script_id` WHERE `user_scripts`.`user_id` = 2 AND `scripts`.`hidden` = 0 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc  [["user_id", 2], ["hidden", false]]
  Rendered shared/_courseblock_action.haml (13.0ms)
  Rendered shared/_course_tall_block.haml (18.8ms)
  *** UserScript Load (0.5ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 41  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
  CACHE (0.2ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  CACHE (0.1ms)  SELECT `scripts`.* FROM `scripts` INNER JOIN `user_scripts` ON `scripts`.`id` = `user_scripts`.`script_id` WHERE `user_scripts`.`user_id` = 2 AND `scripts`.`hidden` = 0 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc  [["user_id", 2], ["hidden", false]]
  Rendered shared/_courseblock_action.haml (13.4ms)
  Rendered shared/_course_tall_block.haml (18.1ms)
  *** UserScript Load (0.5ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 42  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
  *** UserLevel Load (0.4ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 42  ORDER BY id desc
  CACHE (0.1ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  CACHE (0.1ms)  SELECT `scripts`.* FROM `scripts` INNER JOIN `user_scripts` ON `scripts`.`id` = `user_scripts`.`script_id` WHERE `user_scripts`.`user_id` = 2 AND `scripts`.`hidden` = 0 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc  [["user_id", 2], ["hidden", false]]
  CACHE (0.1ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 42  ORDER BY id desc  [["user_id", 2], ["script_id", 42]]
  Rendered shared/_courseblock_action.haml (32.9ms)
  Rendered shared/_course_tall_block.haml (37.8ms)
  *** UserScript Load (0.6ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 1  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
  *** UserLevel Load (0.5ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 1  ORDER BY id desc
  CACHE (0.1ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  CACHE (0.2ms)  SELECT `scripts`.* FROM `scripts` INNER JOIN `user_scripts` ON `scripts`.`id` = `user_scripts`.`script_id` WHERE `user_scripts`.`user_id` = 2 AND `scripts`.`hidden` = 0 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc  [["user_id", 2], ["hidden", false]]
  CACHE (0.2ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 1  ORDER BY id desc  [["user_id", 2], ["script_id", 1]]
  Rendered shared/_courseblock_action.haml (30.2ms)
  Rendered shared/_course_wide_block.haml (35.7ms)
  *** UserScript Load (0.4ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 91  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
  CACHE (0.1ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  CACHE (0.2ms)  SELECT `scripts`.* FROM `scripts` INNER JOIN `user_scripts` ON `scripts`.`id` = `user_scripts`.`script_id` WHERE `user_scripts`.`user_id` = 2 AND `scripts`.`hidden` = 0 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc  [["user_id", 2], ["hidden", false]]
  Rendered shared/_courseblock_action.haml (17.5ms)
  Rendered shared/_course_tall_block.haml (22.9ms)
  *** UserScript Load (0.5ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 89  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
  CACHE (0.3ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 89  ORDER BY id desc  [["user_id", 2], ["script_id", 89]]
  CACHE (0.2ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  CACHE (0.1ms)  SELECT `scripts`.* FROM `scripts` INNER JOIN `user_scripts` ON `scripts`.`id` = `user_scripts`.`script_id` WHERE `user_scripts`.`user_id` = 2 AND `scripts`.`hidden` = 0 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc  [["user_id", 2], ["hidden", false]]
  CACHE (0.2ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 89  ORDER BY id desc  [["user_id", 2], ["script_id", 89]]
  Rendered shared/_courseblock_action.haml (41.5ms)
  Rendered shared/_course_tall_block.haml (46.0ms)
  *** UserScript Load (0.8ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 49  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
  CACHE (0.1ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  CACHE (0.1ms)  SELECT `scripts`.* FROM `scripts` INNER JOIN `user_scripts` ON `scripts`.`id` = `user_scripts`.`script_id` WHERE `user_scripts`.`user_id` = 2 AND `scripts`.`hidden` = 0 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc  [["user_id", 2], ["hidden", false]]
  Rendered shared/_courseblock_action.haml (16.4ms)
  Rendered shared/_course_tall_block.haml (24.3ms)
  *** UserScript Load (0.4ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 51  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
  *** UserLevel Load (0.4ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 51  ORDER BY id desc
  CACHE (0.1ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  CACHE (0.1ms)  SELECT `scripts`.* FROM `scripts` INNER JOIN `user_scripts` ON `scripts`.`id` = `user_scripts`.`script_id` WHERE `user_scripts`.`user_id` = 2 AND `scripts`.`hidden` = 0 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc  [["user_id", 2], ["hidden", false]]
  CACHE (0.1ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 51  ORDER BY id desc  [["user_id", 2], ["script_id", 51]]
  Rendered shared/_courseblock_action.haml (30.8ms)
  Rendered shared/_course_tall_block.haml (35.3ms)
  *** UserScript Load (0.4ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 6  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
  CACHE (0.1ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  CACHE (0.1ms)  SELECT `scripts`.* FROM `scripts` INNER JOIN `user_scripts` ON `scripts`.`id` = `user_scripts`.`script_id` WHERE `user_scripts`.`user_id` = 2 AND `scripts`.`hidden` = 0 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc  [["user_id", 2], ["hidden", false]]
  Rendered shared/_courseblock_action.haml (12.3ms)
  Rendered shared/_course_tall_block.haml (17.4ms)
  *** UserScript Load (0.4ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 52  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
  CACHE (0.1ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  CACHE (0.1ms)  SELECT `scripts`.* FROM `scripts` INNER JOIN `user_scripts` ON `scripts`.`id` = `user_scripts`.`script_id` WHERE `user_scripts`.`user_id` = 2 AND `scripts`.`hidden` = 0 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc  [["user_id", 2], ["hidden", false]]
  Rendered shared/_courseblock_action.haml (13.2ms)
  Rendered shared/_course_tall_block.haml (18.0ms)
  *** UserScript Load (0.4ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 56  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
  CACHE (0.1ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  CACHE (0.1ms)  SELECT `scripts`.* FROM `scripts` INNER JOIN `user_scripts` ON `scripts`.`id` = `user_scripts`.`script_id` WHERE `user_scripts`.`user_id` = 2 AND `scripts`.`hidden` = 0 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc  [["user_id", 2], ["hidden", false]]
  Rendered shared/_courseblock_action.haml (12.5ms)
  Rendered shared/_course_tall_block.haml (17.1ms)
  *** UserScript Load (0.4ms)  SELECT  `user_scripts`.* FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 AND `user_scripts`.`script_id` = 37  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc LIMIT 1
  *** UserLevel Load (0.4ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 37  ORDER BY id desc
  CACHE (0.1ms)  SELECT  1 AS one FROM `user_scripts` WHERE `user_scripts`.`user_id` = 2 LIMIT 1  [["user_id", 2]]
  CACHE (0.1ms)  SELECT `scripts`.* FROM `scripts` INNER JOIN `user_scripts` ON `scripts`.`id` = `user_scripts`.`script_id` WHERE `user_scripts`.`user_id` = 2 AND `scripts`.`hidden` = 0 AND (user_scripts.completed_at is null)  ORDER BY -completed_at asc, greatest(coalesce(started_at, 0), coalesce(assigned_at, 0), coalesce(last_progress_at, 0)) desc, user_scripts.id asc  [["user_id", 2], ["hidden", false]]
  CACHE (0.1ms)  SELECT `user_levels`.* FROM `user_levels` WHERE `user_levels`.`user_id` = 2 AND `user_levels`.`script_id` = 37  ORDER BY id desc  [["user_id", 2], ["script_id", 37]]
  Rendered shared/_courseblock_action.haml (23.3ms)
  Rendered shared/_course_tall_block.haml (28.1ms)
  Rendered home/_all_courses.html.haml (372.8ms)
  Rendered projects/_links.html.haml (6.9ms)
  Rendered home/_projects.html.haml (20.4ms)
  *** (0.8ms)  SELECT COUNT(count_column) FROM (SELECT  1 AS count_column FROM `gallery_activities` WHERE `gallery_activities`.`user_id` = 2 LIMIT 5 OFFSET 0) subquery_for_count
  Rendered home/index.html.haml within layouts/application (1352.4ms)
```

Gatekeeper flag 'logged_in_studio_homepage' set to `false`:

``` bash
Started GET "/" for 127.0.0.1 at 2015-12-04 20:28:48 -0800
Processing by HomeController#index as HTML
  *** User Load (0.6ms)  SELECT  `users`.* FROM `users` WHERE `users`.`deleted_at` IS NULL AND `users`.`id` = 2  ORDER BY `users`.`id` ASC LIMIT 1
  Rendered layouts/_survey.html.haml (0.3ms)
  Rendered devise/confirmations/_alert.html.haml (3.9ms)
  Rendered shared/_courseblock_action.haml (0.4ms)
  Rendered shared/_course_tall_block.haml (5.6ms)
  Rendered shared/_courseblock_action.haml (0.2ms)
  Rendered shared/_course_tall_block.haml (4.7ms)
  Rendered shared/_courseblock_action.haml (0.2ms)
  Rendered shared/_course_tall_block.haml (4.5ms)
  Rendered shared/_courseblock_action.haml (0.2ms)
  Rendered shared/_course_tall_block.haml (6.0ms)
  Rendered shared/_courseblock_action.haml (0.2ms)
  Rendered shared/_course_wide_block.haml (5.9ms)
  Rendered shared/_courseblock_action.haml (0.2ms)
  Rendered shared/_course_tall_block.haml (4.8ms)
  Rendered shared/_courseblock_action.haml (0.2ms)
  Rendered shared/_course_tall_block.haml (4.8ms)
  Rendered shared/_courseblock_action.haml (0.2ms)
  Rendered shared/_course_tall_block.haml (4.8ms)
  Rendered shared/_courseblock_action.haml (0.2ms)
  Rendered shared/_course_tall_block.haml (4.6ms)
  Rendered shared/_courseblock_action.haml (0.2ms)
  Rendered shared/_course_tall_block.haml (5.2ms)
  Rendered shared/_courseblock_action.haml (0.1ms)
  Rendered shared/_course_tall_block.haml (4.5ms)
  Rendered shared/_courseblock_action.haml (0.2ms)
  Rendered shared/_course_tall_block.haml (4.2ms)
  Rendered shared/_courseblock_action.haml (0.1ms)
  Rendered shared/_course_tall_block.haml (4.5ms)
  Rendered home/_all_courses.html.haml (97.2ms)
  Rendered home/index.html.haml within layouts/application (318.4ms)
```

### CODE REVIEW

**Comment by philbogle** [comment]
Thanks Josh.  Great to have this on hand, will consider for Monday deploy.

File: dashboard/app/views/api/user_hero.html.haml [hunk 1]
```diff
@@ -11,23 +8,27 @@
...
-        - if current_page?(root_url)  # Already at studio.code.org
-          #suggestcourse.smalltext= t('home.choose_next_course')
+    - if Gatekeeper.allows('logged_in_studio_hero', default: true)
+      #body{class: role_class}
+        - script = current_user.primary_script
+        - if script
+          - script_level = current_user.next_unpassed_progression_level(script)
+        - if script && script_level
+          #currentprogress
+            = t("data.script.name.#{current_user.primary_script.name}.title")
```
Comment (islemaster) [inline dashboard/app/views/api/user_hero.html.haml:18]: Can replace `current_user.primary_script` with `script`

File: dashboard/app/views/home/index.html.haml [hunk 2]
```diff
@@ -29,17 +29,18 @@
...
+          - course_pair.each do |user_script|
+            = render partial: 'shared/course_progress_block', locals: { script: user_script.script, user_script: user_script }
 
   - else # no logged in user
     = render partial: 'home/sign_in_buttons'
 
   = render partial: 'home/all_courses'
 
-  - if current_user && !current_user.teacher?
+  - if current_user && !current_user.teacher? && Gatekeeper.allows('logged_in_studio_homepage', default: true)
```
Comment (islemaster) [inline dashboard/app/views/home/index.html.haml:43]: Is it slightly better to put the Gatekeeper check first here? (would `current_user.teacher?` ever cause a query?)

File: dashboard/app/views/home/index.html.haml [hunk 3]
```diff
@@ -59,7 +60,7 @@
             -else
               = raw current_user.teachers.map{|t| t[:name]}.join(",")
 
-  - if current_user
+  - if current_user && Gatekeeper.allows('logged_in_studio_homepage', default: true)
```
Comment (islemaster) [inline dashboard/app/views/home/index.html.haml:63]: Ditto ordering here.

**Comment by islemaster** [comment]
LGTM.

**Pull request merged**